File: Python/3541_find_most_frequent_vowel_and_consonant.py
from collections import Counter


class Solution:
    def maxFreqSum(self, s: str) -> int:
        counts = Counter(s)
        vowel_max = 0
        con_max = 0
        for letter, count in counts.items():
            if letter in ['a', 'e', 'i', 'o', 'u'] and count > vowel_max:
                vowel_max = count
            elif letter not in ['a', 'e', 'i', 'o', 'u'] and count > con_max:
                con_max = count
        return vowel_max + con_max

File: JavaScript/3541_find_most_frequent_vowel_and_consonant.js
/**
 * @param {string} s
 * @return {number}
 */
var maxFreqSum = function(s) {
    let letterCounts = new Map();
    for (char of s) {
        if (!letterCounts.has(char)) {
            letterCounts.set(char, 0);
        }
        letterCounts.set(char, letterCounts.get(char) + 1);
    }

    let vowelMax = 0;
    let consonantMax = 0;
    for ([char, count] of letterCounts) {
        if ("aeiou".includes(char)) {
            vowelMax = Math.max(vowelMax, count);
        } else {
            consonantMax = Math.max(consonantMax, count);
        }
    }
    return vowelMax + consonantMax;
};

